security/tpm/test/_tpm_test.js

```diff
@@ -36,21 +41,24 @@ describe('tpm', function () {
   describe('getEndorsementKey', function() {
     it('returns the endorsement key', function(done) {
       /*Tests_SRS_NODE_TPM_SECURITY_CLIENT_06_006: [The `getEndorsementKey` function shall query the TPM hardware and return the `endorsementKey` in the callback.] */
-      var client = new TpmSecurityClient(undefined, fakeSimpleTpmClient );
-      var createPersistStub = sinon.stub(client,'_createPersistentPrimary');
-      var readPersistStub = sinon.stub(client,'_readPersistentPrimary');
-      createPersistStub.withArgs('EK').callsArgWith(4, null, TpmSecurityClient._ekTemplate);
-      createPersistStub.withArgs('SRK').callsArgWith(4, null, TpmSecurityClient._srkTemplate);
-      readPersistStub.withArgs('IDENTITY').callsArgWith(2, null, TpmSecurityClient._srkTemplate)
+      var client = new TpmSecurityClient('', fakeTpm );
+      var readPublicStub = sinon.stub(fakeTpm, 'ReadPublic');
+      var connectStub = sinon.stub(fakeTpm, 'connect');
+      var lastResponseStub = sinon.stub(fakeTpm, 'getLastResponseCode');
+      connectStub.callsArgWith(0);
+      lastResponseStub.returns(tss.TPM_RC.SUCCESS);
+      readPublicStub.withArgs(TpmSecurityClient._ekPersistentHandle).callsArgWith(1, fakeEkKey);
+      readPublicStub.withArgs(TpmSecurityClient._srkPersistentHandle).callsArgWith(1, fakeSrkKey);
+      readPublicStub.withArgs(TpmSecurityClient._idKeyPersistentHandle).callsArgWith(1, fakeIdKey);
```

Comment:
same for this

Reply:
Ok.  Not sure I like it bettert.

---
In reply to: [166799624](https://github.com/Azure/azure-iot-sdk-node/pull/225#discussion_r166799624) [](ancestors = 166799624)